### ISSUE TITLE
`cidr_block` is not a valid option anymore.. it's `ipv4_cidr_block` now

### ISSUE THREAD

**Issue by eldios** [cloudposse/terraform-aws-vpc#121]
### Describe the Bug

The option:
```
cidr_block         = "10.0.0.0/16"
```
is not valid anymore.. it has been changed to `ipv4_cidr_block` now and it's a list of strings, not one string.

### Expected Behavior

```
cidr_block         = "10.0.0.0/16"
```
should work as per docs and examples

### Steps to Reproduce

try to use latest `main` branch version of the module

### Environment

- Module: latest `main` branch commit
- TF version: tested with v1.4.5

**Comment by MajinBuuOnSecurity** [comment]
Easier to make a PR https://github.com/cloudposse/terraform-aws-vpc/pull/123

**Comment by thejrose1984** [comment]
## Findings (code/doc review)

**TL;DR:** The stale `cidr_block = "10.0.0.0/16"` in the README quick-start is real and should be fixed — **but it's not an input to *this* module.** It lives inside the example's `module "dynamic_subnets"` block, so it's an input to the separate `cloudposse/dynamic-subnets/aws` module, which in v2.x renamed `cidr_block` (string) → `ipv4_cidr_block` (`list(string)`). `terraform-aws-vpc`'s own input was never called `cidr_block`; the VPC example already correctly uses `ipv4_primary_cidr_block`. Net: genuine doc bug, slightly misattributed in the title. **Verdict: REPRODUCED** (stale doc example), with two caveats below.

**Environment:** `cloudposse/terraform-aws-vpc` `main` @ `4028a92`.

### What's actually stale

The offending line is in the **dynamic_subnets** example, not the vpc example:

- `README.yaml:112` (source) and `README.md:97` (generated):
  ```hcl
  module "dynamic_subnets" {
    source = "cloudposse/dynamic-subnets/aws"
    ...
    cidr_block = "10.0.0.0/16"   # <- obsolete dynamic-subnets input
  }
  ```
- The `module "vpc"` example just above it (`README.yaml:97` / `README.md:65`) already uses the correct `ipv4_primary_cidr_block = "10.0.0.0/16"` — a `string` (see `variables.tf:1`). So the VPC module itself is fine.

The reporter's description — "changed to `ipv4_cidr_block` … a list of strings" — matches **dynamic-subnets v2.x**, not this module, which is the tell that the stale token belongs to the dynamic_subnets block.

### Corroborating evidence (this repo's own working example)

`examples/vpc-endpoints/main.tf:75-82` pins dynamic-subnets v2.0.4 and uses the **new** name + list form:
```hcl
module "subnets" {
  source          = "cloudposse/dynamic-subnets/aws"
  version         = "2.0.4"
  ...
  ipv4_cidr_block = [module.vpc.vpc_cidr_block]   # list, not string
}
```
So the correct replacement is `ipv4_cidr_block = ["10.0.0.0/16"]` (rename **and** string→list), exactly as the existing PR #123 proposes.

### Caveat 1 — PR #123 will be reverted by the next README regen

PR #123 makes the right content change (`cidr_block = "10.0.0.0/16"` → `ipv4_cidr_block = ["10.0.0.0/16"]`) but edits **only `README.md`**, which is an auto-generated artifact (`README.md:19`: "This file was automatically generated by the `cloudposse/build-harness`"). The source of truth is `README.yaml:112`, which still contains the stale `cidr_block`. The next `make readme` will overwrite PR #123's change. **The fix must be applied to `README.yaml:112`** (then regenerate `README.md`), or to both files in the same PR.

### Caveat 2 — scope

Only this one line is stale; the other `cidr_block` hits in `README.md` are legitimate variable names (`ipv4_*_cidr_block_*`, `ipv6_*_cidr_block_*`) from the auto-generated inputs table, not examples.

### Recommendation
- Update `README.yaml:112` to `ipv4_cidr_block = ["10.0.0.0/16"]` and regenerate `README.md` (`make init && make readme`), so the fix survives regeneration. Ask PR #123's author to retarget `README.yaml`, or supersede it.
- Optional: add a clarifying comment in the example that `ipv4_cidr_block` is a `cloudposse/dynamic-subnets/aws` input (not a `terraform-aws-vpc` input) to prevent the same misattribution.
